mongodbatlas/data_lakes.go

```diff
@@ -0,0 +1,236 @@
+}
+
+type DataLakeDatabase struct {
+	Name        string                 `json:"name,omitempty"`
+	Collections []DataLakeCollection   `json:"collections,omitempty"`
+	Views       []DataLakeDatabaseView `json:"views,omitempty"`
+}
+
+type Storage struct {
+	Databases map[string]DataLakeDatabase `json:"databases,omitempty"`
+	Stores    []DataLakeStore             `json:"stores,omitempty"`
+}
+
+// DataLake represents a data lake.
+type DataLake struct {
+	CloudProviderConfig CloudProviderConfig `json:"cloudProviderConfig,omitempty"` // Configuration for the cloud service where Data Lake source data is stored.
+	DataProcessRegion   DataProcessRegion   `json:"dataProcessRegion,omitempty"`   // Cloud provider region which clients are routed to for data processing.
+	GroupID             string              `json:"groupId,omitempty"`             // Unique identifier for the project.
+	Hostnames           []string            `json:"hostnames,omitempty"`           // List of hostnames for the data lake.
+	Name                string              `json:"name,omitempty"`                // Name of the data lake.
+	State               string              `json:"state,omitempty"`               // Current state of the data lake.
+	Storage             Storage             `json:"storage,omitempty"`             // Configuration for each data store and its mapping to MongoDB collections / databases.
+}
+
+// DataLakeReqPathParameters represents all the possible parameters to make the request
+type DataLakeReqPathParameters struct {
+	GroupID string `json:"groupId,omitempty"` // The unique identifier of the project for the Atlas cluster.
+	Name    string `json:"name,omitempty"`    // The name of the Data Lake.
+}
+
+// DataLakeReqPathParameters represents all possible fields that can be updated in a data lake
+type DataLakeUpdateRequest struct {
+	CloudProviderConfig CloudProviderConfig `json:"cloudProviderConfig,omitempty"`
+	DataProcessRegion   DataProcessRegion   `json:"dataProcessRegion,omitempty"`
+}
+
+// DataLakeReqPathParameters represents the required fields to create a new data lake
+type DataLakeCreateRequest struct {
+	Name string `json:"name,omitempty"`
+}
+
+// List gets all data lakes for the specified group.
+// See more: https://docs.mongodb.com/datalake/reference/api/dataLakes-get-all-tenants
+func (s *DataLakeServiceOp) List(ctx context.Context, requestParameters *DataLakeReqPathParameters, listOptions *ListOptions) ([]DataLake, *Response, error) {
+	if requestParameters.GroupID == "" {
+		return nil, nil, NewArgError("groupId", "must be set")
+	}
+
+	path := fmt.Sprintf("%s/%s/dataLakes", dataLakesBasePath, requestParameters.GroupID)
+
+	path, err := setListOptions(path, listOptions)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new([]DataLake)
+	resp, err := s.Client.Do(ctx, req, root)
```

Comment:
you may want to do 
```suggestion
	var root []DataLake
	resp, err := s.Client.Do(ctx, req, &root)
```
instead, it's a more common idiom for empty slices and also saves you from the `return *root` to just `return root`